client/descriptions.go

```diff
@@ -378,6 +447,36 @@ func (f FieldDescription) IsArray() bool {
 		f.Kind == FieldKind_NILLABLE_STRING_ARRAY
 }
 
+// IsObject returns true if this field is an object type.
+func (f SchemaFieldDescription) IsObject() bool {
+	return (f.Kind == FieldKind_FOREIGN_OBJECT) ||
+		(f.Kind == FieldKind_FOREIGN_OBJECT_ARRAY)
+}
+
+// IsObjectArray returns true if this field is an object array type.
+func (f SchemaFieldDescription) IsObjectArray() bool {
+	return (f.Kind == FieldKind_FOREIGN_OBJECT_ARRAY)
+}
+
+// IsRelation returns true if this field is a relation.
+func (f SchemaFieldDescription) IsRelation() bool {
+	return f.RelationName != ""
+}
+
+// IsArray returns true if this field is an array type which includes inline arrays as well
+// as relation arrays.
+func (f SchemaFieldDescription) IsArray() bool {
+	return f.Kind == FieldKind_BOOL_ARRAY ||
+		f.Kind == FieldKind_INT_ARRAY ||
+		f.Kind == FieldKind_FLOAT_ARRAY ||
+		f.Kind == FieldKind_STRING_ARRAY ||
+		f.Kind == FieldKind_FOREIGN_OBJECT_ARRAY ||
+		f.Kind == FieldKind_NILLABLE_BOOL_ARRAY ||
+		f.Kind == FieldKind_NILLABLE_INT_ARRAY ||
+		f.Kind == FieldKind_NILLABLE_FLOAT_ARRAY ||
+		f.Kind == FieldKind_NILLABLE_STRING_ARRAY
+}
```

Comment:
thought: Another not ideal side effect of this is that we need to dupliucate these methods.

Reply:
That is quite cheap, I'm not worried about that.

Reply:
I don't think you can justify copy&paste by performance characteristics.

I'd say you either make one struct embed the other or create something like `client.IsArrayKind(Kind)`

Reply:
cheap in terms of dev cost, performance is not relevant here

Reply:
`client.IsArrayKind(Kind)` or `Kind.IsArray()` is a nice thought though, can do the same for the others too.

- [ ] Move Kind related funcs